inference_experimental/inference_exp/models/auto_loaders/models_registry.py

```diff
@@ -12,7 +13,15 @@
 EMBEDDING_TASK = "embedding"
 
 
-REGISTERED_MODELS: Dict[Tuple[ModelArchitecture, TaskType, BackendType], LazyClass] = {
+@dataclass(frozen=True)
+class RegistryEntry:
+    model_class: LazyClass
+    supported_model_features: Optional[Set[str]] = field(default=None)
+
+
+REGISTERED_MODELS: Dict[
+    Tuple[ModelArchitecture, TaskType, BackendType], Union[LazyClass, RegistryEntry]
```

Comment:
actually by having it in value makes it so we cant have two entries one with without model_features, but maybe thats ok / would never be needed?